ee/tabby-webserver/src/service/license.rs

```diff
@@ -48,6 +49,51 @@ pub fn validate_license(token: &str) -> Result<LicenseInfo, jwt::errors::ErrorKi
     Ok(data?.claims)
 }
 
+fn jwt_timestamp_to_utc(f: f64) -> Result<DateTime<Utc>> {
+    let secs = f as i64;
+    Ok(NaiveDateTime::from_timestamp_millis(secs * 1000)
+        .ok_or_else(|| anyhow!("Timestamp is corrupt"))?
+        .and_utc())
+}
+
+#[async_trait]
+impl LicenseService for DbConn {
+    async fn get_license_info(&self) -> Result<Option<LicenseInfo>> {
+        let Some(license) = self.read_enterprise_license().await? else {
+            return Ok(None);
+        };
+        let license =
+            validate_license(&license).map_err(|e| anyhow!("License is corrupt: {e:?}"))?;
```

Comment:
You need to handle the `Expire` error generated by `validate_license` for LicenseStatus

Reply:
That's not the case. You could check the `test_expired_token` unit test, which provides an example on how to handle expire error.

Reply:
![image](https://github.com/TabbyML/tabby/assets/14198267/2b02f39f-eca4-4030-b1b7-18202528a964)
This test never gets the license info. The validate function returns an error.

Reply:
ah - so you're worrying about the parsing part.

It can be done by decode the second component of jwt token from base64, then it's just vanilla json. See jwt token specs for details.